src/error.rs

```diff
@@ -86,3 +95,23 @@ impl From<futures::channel::oneshot::Canceled> for ConnectionError {
         ConnectionError::Closed
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn connection_error_can_be_question_marked_to_io_error() {
+        returns_io_error().expect_err("to fail");
+    }
+
+    fn returns_io_error() -> std::io::Result<()> {
+        always_err()?;
+
+        Ok(())
+    }
+
+    fn always_err() -> Result<(), ConnectionError> {
+        Err(ConnectionError::Closed)
+    }
+}
```

Comment:
> `connection_error_can_be_question_marked_to_io_error`

I have to admit that I don't really see the point of this test. It's general rust logic that we can question mark one error into another if `From<>` is implemented. Since we implemented `impl From<ConnectionError> for std::io::Error` we know for sure that this will not fail.
From my understanding, tests should check for implementation errors, edge cases, side effect, ... etc. This is not the case here. What is the gain from this test?

Reply:
I wanted to test that what I wanted to achieve in `rust-libp2p` actually works. Happy to delete the test again if you think it is bloat (I think I agree with you).